frontend/components/transaction/SwapTransaction.vue

```diff
@@ -29,32 +29,49 @@
             </Explain>
             the auction. In case your transaction will be rejected, it only results in the loss of the transaction fee.
         </TextBlock>
-        <WalletBlock
-            class="mb-6 lg:mb-0"
-            :disabled="!auctionTransaction.isActive"
-            :is-loading="isConnecting"
-            :wallet-address="walletAddress"
-            :is-explanations-shown="isExplanationsShown"
-            @connectWallet="$emit('connect')"
-            @disconnectWallet="$emit('disconnect')"
-        />
-        <AuthorisationBlock
-            class="mb-6 lg:mb-0"
-            :disabled="!isConnected || !auctionTransaction.isActive"
-            :auction-transaction="auctionTransaction"
-            :is-loading="isAuthorizing"
-            :is-wallet-authorized="isWalletAuthorized"
-            :is-collateral-authorised="isCollateralAuthorised"
-            :is-explanations-shown="isExplanationsShown"
-            @authorizeWallet="$emit('authorizeWallet')"
-            @authorizeCollateral="$emit('authorizeCollateral', auctionTransaction.collateralType)"
-        />
+        <div>
+            <WalletConnectionCheckPanel
+                :is-correct.sync="isWalletConnectedCheck"
+                :wallet-address="walletAddress"
+                :is-loading="isConnecting"
+                :is-explanations-shown="isExplanationsShown"
+                :disabled="!isAuctionActiveAndNotFinished"
+                @connectWallet="$emit('connect')"
+                @disconnectWallet="$emit('disconnect')"
+            />
+            <WalletAuthorizationCheckPanel
+                :is-correct.sync="isWalletDAIAuthorizationCheckPassed"
+                :is-wallet-authorized="isWalletAuthorized"
+                :disabled="!isAuctionActiveAndNotFinished || !isWalletConnectedCheck"
+                :wallet-address="walletAddress"
+                :is-explanations-shown="isExplanationsShown"
+                :is-loading="isAuthorizing"
+                @authorizeWallet="$emit('authorizeWallet')"
+            />
+            <CollateralAuthorizationCheckPanel
+                :is-correct.sync="isWalletCollateralAuthorizationCheckPassed"
+                :collateral-type="auctionTransaction.collateralType"
+                :is-collateral-authorized="isCollateralAuthorised"
+                :auth-transaction-fee-e-t-h="auctionTransaction.authTransactionFeeETH"
+                :wallet-address="walletAddress"
+                :is-loading="isAuthorizing"
+                :disabled="!isAuctionActiveAndNotFinished || !isWalletDAIAuthorizationCheckPassed"
+                :is-explanations-shown="isExplanationsShown"
+                @authorizeCollateral="$emit('authorizeCollateral', auctionTransaction.collateralType)"
+            />
+            <ProfitCheckPanel
+                :is-correct.sync="isProfitCheckPassed"
```

Comment:
Currently, `isProfitCheckPassed` will be false in case of `The transaction net profit is negative`, but should be true, since (according to the text of the notice) the transaction should be executable

(you have to change `currentStateAndTitle` watcher, but github doesn't let me comment there)